co-simulation/fed/mosaic-sumo/src/main/java/org/eclipse/mosaic/fed/sumo/ambassador/AbstractSumoAmbassador.java

```diff
@@ -1229,6 +1242,7 @@ public synchronized void processTimeAdvanceGrant(long time) throws InternalFeder
                 rti.triggerInteraction(simulationStepResult.getTrafficDetectorUpdates());
                 this.rti.triggerInteraction(simulationStepResult.getTrafficLightUpdates());
                 receivedSimulationStep = false;
+                firstTimePrintingTime = true;
```

Comment:
Why do you need both of these variables. `receivedSimulationStep` should be `!firstTimePrintingTime`

Reply:
Because they are meant for different cases. 
`receivedSimulationStep` is flagged when `carla` returns with some handshake to sumoAmbassador. However, sumoAmbassador is called many times before that by MOSAIC. 
`firstTimePrintingTime` is meant to track the first time ever sumoAmbassador is called to progressTime. And that is by design. 
In other words, we can't reuse the same variable, because if we pick `!receivedSimulationStep`, the log will still end up printing many times in almost like 1ms apart. This `firstTimePrintingTime` is meant to avoid such excessive logging.